src/main/webapp/app/exercises/programming/manage/update/update-components/custom-build-plans/programming-exercise-build-configuration/programming-exercise-build-configuration.component.ts

```diff
@@ -1,21 +1,77 @@
-import { Component, input, output, viewChild } from '@angular/core';
+import { Component, OnInit, effect, input, output, viewChild } from '@angular/core';
 import { NgModel } from '@angular/forms';
+import { ProgrammingExercise, ProgrammingLanguage } from 'app/entities/programming/programming-exercise.model';
+
+const ALLOWED_DOCKER_FLAG_OPTIONS = ['network', 'env'];
+
+const NOT_SUPPORTED_NETWORK_DISABLED_LANGUAGES = [ProgrammingLanguage.SWIFT, ProgrammingLanguage.HASKELL, ProgrammingLanguage.EMPTY];
 
 @Component({
     selector: 'jhi-programming-exercise-build-configuration',
     templateUrl: './programming-exercise-build-configuration.component.html',
     styleUrls: ['../../../../programming-exercise-form.scss'],
 })
-export class ProgrammingExerciseBuildConfigurationComponent {
+export class ProgrammingExerciseBuildConfigurationComponent implements OnInit {
+    programmingExercise = input<ProgrammingExercise>();
     dockerImage = input.required<string>();
     dockerImageChange = output<string>();
 
     timeout = input<number>();
     timeoutChange = output<number>();
 
+    envVars: string = '';
+    isNetworkDisabled = false;
+
     isAeolus = input.required<boolean>();
 
     dockerImageField = viewChild<NgModel>('dockerImageField');
-
     timeoutField = viewChild<NgModel>('timeoutField');
+
+    isLanguageSupported: boolean = false;
+
+    readonly NETWORK_KEY: string = 'network';
+    readonly ENV_KEY: string = 'env';
+    readonly ENV_VAR_REGEX = /(?:'([^']+)'|"([^"]+)"|(\w+))=(?:'([^']*)'|"([^"]*)"|([^,]+))/;
```

Comment:
I'm not convinced that this is the best regex for what you are trying to achieve here. It e.g. matches `"abc=42` which is probably not intended.
I think the non-capturing groups are not necessary here, and overcomplicate things.

What should the inputed envVars look like? `ENV_VAR=15` / `ENV_VAR="15"` / ... should lower case be allowed?
A very simple regex would be ` ^[A-Z_][A-Z0-9_]*="[^"]*"$ ` 

Maybe even add a comment above, shortly defining in prose which env var formatting you expect

Reply:
`^[A-Z_][A-Z0-9_]*="[^"]*"$` wouldn't be valid as it misses commas that are required for multiple vars. 
It is up to the instructor to check the validity of the keys, I just wanted to give them the options to enter the vars in formats they would be familiar with (i.e. allowing single and double quotes and allowing spaces in keys/values)
So `"abc=42` is technically not a valid env var in Linux, but it's the instructors' responsibility

Reply:
But if it's the instructors responsibility anyway, why do we restrict it at all?